sprint_mision_6/sprint-mission-6.sql

```diff
@@ -0,0 +1,311 @@
+-- # 초급 문제
+-- 1. `orders` 테이블에서 모든 주문을 조회하세요.
+SELECT
+    *
+FROM
+    orders;
+
+-- 2. `orders`테이블에서 `id` 가 `423`인 주문을 조회하세요.
+SELECT
+    *
+FROM
+    orders
+WHERE
+    id = 423;
+
+-- 3. `orders` 테이블에서 총 주문 건수를 `total_orders`라는 이름으로 구하세요.
+SELECT
+    COUNT(id) AS total_orders
+FROM
+    orders;
+
+-- 4. `orders` 테이블에서 최신 순으로 주문을 조회하세요. (`date`, `time` 컬럼이 분리되어 있다는 점에 주의)
+SELECT
+    *
+FROM
+    orders
+ORDER BY
+    date DESC,
+    time DESC;
+
+-- 5. `orders` 테이블에서 오프셋 기반 페이지네이션된 목록을 조회합니다. 페이지 크기가 10이고 최신순일 때, 첫 번째 페이지를 조회하세요.
+SELECT
+    *
+FROM
+    orders
+ORDER BY
+    date DESC,
+    time DESC
+LIMIT
+    10 OFFSET 0;
+
+-- 6. `orders` 테이블에서 오프셋 기반 페이지네이션된 목록을 조회합니다. 페이지 크기가 10이고 최신순일 때 5번째 페이지를 조회하세요.
+SELECT
+    *
+FROM
+    orders
+ORDER BY
+    date DESC,
+    time DESC
+LIMIT
+    10 OFFSET 40;
+
+-- 7. `orders` 테이블에서 커서 페이지네이션된 목록을 조회합니다. 페이지 크기가 10이고 최신순일때, `id` 값을 기준으로 커서를 사용합시다. 커서의 값이 `42`일 때 다음 페이지를 조회하세요.
+SELECT
+    *
+FROM
+    orders
+WHERE
+    id < 42
+ORDER BY
+    id DESC
+LIMIT
+    10;
+
+-- 8. `orders` 테이블에서 2025년 3월에 주문된 내역만 조회하세요.
+SELECT
+    *
+FROM
+    orders
+WHERE
+    date BETWEEN '2025-03-01'
+    AND '2025-03-31';
+
+-- 9. `orders` 테이블에서 2025년 3월 12일 오전에 주문된 내역만 조회하세요.
+SELECT
+    *
+FROM
+    orders
+WHERE
+    date = '2025-03-12'
+    AND time >= '00:00:00'
+    AND time < '12:00:00';
+
+-- 10. `pizza_types` 테이블에서 이름에 'Cheese' 혹은 'Chicken'이 포함된 피자 종류를 조회하세요. (대소문자를 구분합니다)
+SELECT
+    *
+FROM
+    pizza_types
+WHERE
+    name LIKE '%Cheese%'
+    OR name LIKE '%Chicken%';
+
+-- # 중급 문제
+-- 1. `order_details` 테이블에서 각 피자(`pizza_id`)별로 주문된 건 수(`order_id`)를 보여주세요.
+SELECT
+    pizza_id,
+    COUNT(*)
+FROM
+    order_details
+GROUP BY
+    pizza_id;
```

Comment:
`주문된 건 수`라는건 서로 다른 order_id를 보여줘야만할 것 같은데,
order_details에 동일 주문이 같은 pizza_id로 여러 라인일 수 있는 경우를 고려해야해요.

```suggestion
SELECT
    pizza_id,
    COUNT(DISTINCT order_id) as order_count
FROM
    order_details
GROUP BY
    pizza_id;
```